recipes/boost/all/conanfile.py

```diff
@@ -855,103 +1001,194 @@ def _is_versioned_layout(self):
+
+                self.cpp_info.components[module].requires = self._dependencies["dependencies"][module] + ["_libboost"]
+                self.cpp_info.components[module].names["cmake_find_package"] = module
+                self.cpp_info.components[module].names["cmake_find_package_multi"] = module
+
+                for requirement in self._dependencies.get("requirements", {}).get(module, []):
+                    if requirement == "backtrace":
+                        # FIXME: backtrace not (yet) available in cci
+                        continue
+                    if self.options.get_safe(requirement, None) == False:
+                        continue
+                    conan_requirement = self._option_to_conan_requirement(requirement)
+                    if not conan_requirement:
+                        continue
+                    if conan_requirement in ("icu", "iconv"):
+                        if conan_requirement != self.options.get_safe("i18n_backend"):
+                            continue
+                    self.cpp_info.components[module].requires.append("{0}::{0}".format(conan_requirement))
+
+        if used_libraries != detected_libraries:
+            non_used = detected_libraries.difference(used_libraries)
+            assert len(non_used) == 0, "These libraries were not used in conan components: {}".format(non_used)
+
+            non_existing = used_libraries.difference(detected_libraries)
+            assert len(non_existing) == 0, "These libraries were used, but not built: {}".format(non_existing)
+
+        # Apply these options to the 'headers' component so header_only has these also applied
         if not self.options.header_only and self.options.shared:
-            self.cpp_info.defines.append("BOOST_ALL_DYN_LINK")
+            self.cpp_info.components["headers"].defines.append("BOOST_ALL_DYN_LINK")
 
         if self.options.system_no_deprecated:
-            self.cpp_info.defines.append("BOOST_SYSTEM_NO_DEPRECATED")
+            self.cpp_info.components["headers"].defines.append("BOOST_SYSTEM_NO_DEPRECATED")
 
         if self.options.asio_no_deprecated:
-            self.cpp_info.defines.append("BOOST_ASIO_NO_DEPRECATED")
+            self.cpp_info.components["headers"].defines.append("BOOST_ASIO_NO_DEPRECATED")
 
         if self.options.filesystem_no_deprecated:
-            self.cpp_info.defines.append("BOOST_FILESYSTEM_NO_DEPRECATED")
+            self.cpp_info.components["headers"].append("BOOST_FILESYSTEM_NO_DEPRECATED")
 
         if self.options.segmented_stacks:
-            self.cpp_info.defines.extend(["BOOST_USE_SEGMENTED_STACKS", "BOOST_USE_UCONTEXT"])
-
-        if self.settings.os != "Android":
-            if self._gnu_cxx11_abi:
-                self.cpp_info.defines.append("_GLIBCXX_USE_CXX11_ABI=%s" % self._gnu_cxx11_abi)
+            self.cpp_info.components["headers"].extend(["BOOST_USE_SEGMENTED_STACKS", "BOOST_USE_UCONTEXT"])
```

Comment:
Likewise, `.defines.extend()`?
```suggestion
            self.cpp_info.components["headers"].defines.extend(["BOOST_USE_SEGMENTED_STACKS", "BOOST_USE_UCONTEXT"])
```